app/uk/gov/hmrc/rasapi/repository/RasFileRepository.scala

```diff
@@ -15,116 +15,115 @@
     log.info("[RasFileRepository][saveFile] Starting to save file")
 
-    gridFSG.writeFromInputStream(generateFileToSave(fileId, contentType, envelopeId, userId), new FileInputStream(filePath.toFile)).map{ res =>
-      log.warn(s"Saved File id is ${res.id} for userId ($userId)")
-      res
-    }.recover{ case ex: Throwable =>
-        log.error(s"error saving file -> $fileId for userId ($userId). Exception: ${ex.getMessage}")
-        throw new RuntimeException(s"failed to save file due to error" + ex.getMessage)
+    val observableToUploadFrom: Observable[ByteBuffer] = Observable(
+      Seq(ByteBuffer.wrap(Files.readAllBytes(filePath).array)
+    ))
+
+    val options: GridFSUploadOptions = new GridFSUploadOptions()
+      .metadata(Document(
+        "envelopeId" -> envelopeId,
+        "fileId" -> fileId,
+        "userId" -> userId,
+        "contentType" -> contentType))
+
+    gridFSG.uploadFromObservable(fileId, observableToUploadFrom, options).head().flatMap { res =>
+      log.warn(s"[RasFileRepository][saveFile] Saved file $fileId for user $userId")
+      checkAndEnsureTTL(mongoComponent.database, s"$bucketName.files")
+      gridFSG.find(Document("_id" -> res)).head()
+    }.recover {
+      case e: Throwable =>
+        log.error(s"[RasFileRepository][saveFile] Error saving the file $fileId for user $userId. Exception: ${e.getMessage}")
+        throw new RuntimeException(s"Failed to save file due to error: ${e.getMessage}")
     }
   }
 
   def fetchFile(_fileName: String, userId: String)(implicit ec: ExecutionContext): Future[Option[FileData]] = {
-      log.info(s"id in repo input is ${_fileName} for userId ($userId).")
-
-      gridFSG.find[BSONDocument, ResultsFile](BSONDocument("filename" -> _fileName)).headOption.map {
-      case Some(file) =>   log.info(s"file fetched ${file.id} file size = ${file.length}")
-        Some(FileData(file.length, gridFSG.enumerate(file)))
-      case None => log.warn(s"file not found for userId ($userId).")
-        None
-    }.recover{
-      case ex:Throwable =>
-        log.error(s"error trying to fetch file ${_fileName} for userId ($userId). Exception: ${ex.getMessage}")
-        throw new RuntimeException("failed to fetch file due to error" + ex.getMessage)
+    log.info(s"[RasFileRepository][fetchFile] Attempting to fetch file ${_fileName} for userId ($userId).")
+    gridFSG.find(Document("filename" -> _fileName)).headOption().flatMap {
+      case file @Some(_) =>
+        log.info(s"[RasFileRepository][fetchFile] Found ${_fileName} for userId ($userId).")
+        val fileCombined: Array[Byte] = Await.result(gridFSG.downloadToObservable(file.get.getObjectId).toFuture(), Duration(10, SECONDS)).map(bb => bb.array).reduceLeft(_ ++ _)
+        log.info(s"[RasFileRepository][fetchFile] Successfully downloaded ${_fileName} for userId ($userId).")
+        Future(Some(FileData(file.get.getLength, Enumerator(fileCombined))))
```

Comment:
since this is a Future, I think you might be able to remove that Await on line 82 (e.g. by `andThen`ning it?)

Reply:
Not entirely sure what you mean here - might be worth explaining on call?